pkg/importer/util.go

```diff
@@ -52,3 +54,28 @@ func StreamDataToFile(dataReader io.Reader, filePath string) error {
 	}
 	return nil
 }
+
+// StreamDataToBlockDevice provides a function to stream the specified io.Reader to the specified block device
+func StreamDataToBlockDevice(dataReader io.Reader, filePath string) error {
+	var fd int
+	var err error
+
+	fd, err = syscall.Open(filePath, os.O_CREATE|os.O_WRONLY, 0777)
+	defer syscall.Close(fd)
+	if err != nil {
+		return errors.Wrapf(err, "could not open device file %q", filePath)
+	}
+
+	buf := new(bytes.Buffer)
+	glog.V(1).Infof("reading from dataReader....\n")
+	buf.ReadFrom(dataReader)
+	glog.V(1).Infof("writing to block device....\n")
+	if _, err = syscall.Write(fd, buf.Bytes()); err != nil {
+		syscall.Unlink(filePath)
```

Comment:
Unlinking a block device?  Why?

Reply:
Because we create the destination file ourselves.  We don't create the destination block device.